crates/orderbook/src/fee.rs

```diff
@@ -281,27 +282,29 @@ impl MinFeeCalculator {
         &self,
         fee_data: FeeData,
     ) -> Result<FeeParameters, PriceEstimationError> {
-        let gas_price = self.gas_estimator.estimate().await?.effective_gas_price();
-        let gas_amount = self
-            .price_estimator
-            .estimate(&price_estimation::Query {
-                sell_token: fee_data.sell_token,
-                buy_token: fee_data.buy_token,
-                in_amount: fee_data.amount,
-                kind: fee_data.kind,
-            })
-            .await?
-            .gas
-            .to_f64_lossy();
-        let fee_in_eth = gas_price * gas_amount;
-        let query = price_estimation::Query {
+        let buy_token_query = price_estimation::Query {
+            sell_token: fee_data.sell_token,
+            buy_token: fee_data.buy_token,
+            in_amount: fee_data.amount,
+            kind: fee_data.kind,
+        };
+        let native_token_query = price_estimation::Query {
             sell_token: fee_data.sell_token,
             buy_token: self.native_token,
             in_amount: self.native_token_price_estimation_amount,
             kind: OrderKind::Buy,
         };
-        let estimate = self.price_estimator.estimate(&query).await?;
-        let sell_token_price = estimate.price_in_sell_token_f64(&query);
+        let (gas_estimate, buy_token_estimate, native_token_estimate) = try_join!(
+            self.gas_estimator
+                .estimate()
+                .map_err(PriceEstimationError::from),
```

Comment:
Do we need to explicitly call `map_err` here? Previously we did not.

Reply:
`try_join` expects the same return type for all futures and `gas_estimator.estimate` returns and anyhow error whereas the other one returns a PriceEstimationError (and the compiler doesn't seem smart enough to auto convert the type.